libpod/in_memory_state.go

```diff
@@ -332,6 +355,89 @@ func (s *InMemoryState) RemovePod(pod *Pod) error {
 	return nil
 }
 
+// UpdatePod updates a pod's state from the backing database
+// As in-memory states have no database this is a no-op
+func (s *InMemoryState) UpdatePod(pod *Pod) error {
+	return nil
+}
+
+// AddContainerToPod adds a container to the given pod, also adding it to the
+// state
+func (s *InMemoryState) AddContainerToPod(pod *Pod, ctr *Container) error {
+	if !pod.valid {
+		return errors.Wrapf(ErrPodRemoved, "pod %s is not valid and cannot be added to", pod.ID())
```

Comment:
Would ErrNoSuchPod be better here?  Or perhaps a new error, ErrInvalidPod?  Or are we pretty certain that the only time we'd get here is if the pod had been removed first?  Similar comment for line 371.

Reply:
We only flip Valid to false if a container/pod no longer exists in the database, so it's a safe assumption that an invalid container or pod has been removed